addons/website_slides/static/src/js/slides_slide_quiz_add.js

```diff
@@ -0,0 +1,428 @@
+odoo.define('website_slides.quiz_creation', function (require) {
+    'use strict';
+
+    var core = require('web.core');
+    var publicNotification = require('web.Notification');
+    var publicWidget = require('web.public.widget');
+    var QWeb = core.qweb;
+    var _t = core._t;
+
+    var QuestionDeletionNotification = publicNotification.extend({
+
+        _autoCloseDelay: 5000,
+
+        init: function (parent, params) {
+            var self = this;
+            this._super(parent, params);
+            this.sticky = true;
+            this.icon = 'fa-trash';
+            this.className += ' bg-danger';
+            this.question = params.question;
+            this.continueDeletion = true;
+            this.buttons = [
+                {
+                    text: _t('Undo'),
+                    primary: true,
+                    click: function () {
+                        self.continueDeletion = false;
+                    }
+                }
+            ]
+        },
+
+        start: function() {
+            this._super();
+            this._initTimer();
+        },
+
+        _initTimer: function() {
+            var i = 4;
+            var self = this;
+            this.timer = setInterval(function () {
+                self.$el.find('div.o_notification_content').html(_.str.sprintf(_t("Click Undo to cancel, or in " +
+                    "%s seconds the question \"%s\" will be deleted"), i--, self.question.data('title')));
+            }, 1000);
+            setTimeout(function () {
+                self.close();
+            }, 5000)
+        },
+
+        _onClose: function(ev) {
+            console.log('onClose method');
+            if (this.continueDeletion) {
+                core.bus.trigger('onDeleteQuestion', this.question);
+            } else {
+                core.bus.trigger('onCancelDeletion', this.question);
+            }
+            clearInterval(this.timer);
+            this._super(ev);
+        }
+    });
+
+    var NewQuestionWidget = publicWidget.Widget.extend({
```

Comment:
Looking at interface / code, it seems code could be improved by separating a bit more question / manager. For example

 -> QuestionWidget: handle a given question (validate answers, check valid answer, name, reset answer, delete it)
 -> QuizManager: handle questions (meaning for example _onAddQuestionClick should trigger a new QuestionWidget instantiated by the Manager)

Currently code seems a bit complicated, it is just a proposal to separate question management from quiz management :) . Do you think this proposal could help ?